### PR TITLE
DOC: Add PyArrow user guide

### PR DESCRIPTION
**Pull request by mroeschke**

### CODE REVIEW

**Review by MarcoGorelli — APPROVED** [review]
Nice, this'll be useful - I've left a really minor comment

File: doc/source/user_guide/pyarrow.rst [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+   import io
+   data = io.StringIO("""a,b,c
+      1,2.5,True
+      3,4.5,False
+   """)
+   df = pd.read_csv(data, engine="pyarrow")
+   df
+
+By default, these functions, and all other IO reader functions, return NumPy-backed data. These readers can return
+PyArrow-backed data by specifying ``use_nullable_dtypes`` with the global configuration option ``"mode.dtype_backend"``
```
Comment (phofl) [inline doc/source/user_guide/pyarrow.rst:123]: Could you reference the global option to set ``use_nullable_dtypes``as well?

**Review by phofl — COMMENTED** [review]
small comment, otherwise lgtm

File: doc/source/user_guide/pyarrow.rst [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+*********************
+PyArrow Functionality
+*********************
+
+pandas can utilize `PyArrow <https://arrow.apache.org/docs/python/index.html>`__ to extend functionality and improve the performance
+of various APIs. This includes:
+
+* More extensive `data types <https://arrow.apache.org/docs/python/api/datatypes.html>`__ compared to NumPy
+* Missing data support (NA) for all data types
+* Performant IO reader integration
```
Comment (datapythonista) [inline doc/source/user_guide/pyarrow.rst:14]: Do you think is it work to mention interoperability (e.g. export data to polars)? Kind of similar to the last point, but IO reader seems to be more about reading parquet or csv files.

File: doc/source/user_guide/pyarrow.rst [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+therefore, supported functionality exists where this interface is integrated within the pandas API. Additionally, this functionality
+is accelerated with PyArrow `compute functions <https://arrow.apache.org/docs/python/api/compute.html>`__ where available. This includes:
+
+* Numeric aggregations
+* Numeric arithmetic
+* Numeric rounding
+* Logical and comparison functions
+* String functionality
+* Datetime functionality
+
```
Comment (datapythonista) [inline doc/source/user_guide/pyarrow.rst:81]: Maybe worth adding that below are just some examples of native Arrow-backed functions? I feel a bit of context is missing to understand what you're trying to show with those examples.

File: doc/source/user_guide/pyarrow.rst (hunk 1 above)
Comment (datapythonista) [reply doc/source/user_guide/pyarrow.rst:123]: I find this sentence a bit confusing, seems like we're first suggesting to use `mode.dtype_backend='use_nullable_dtypes'`. Maybe we can write it like `by specifying the parameter use nullable_dtypes *and* the global configuration ...`, or something else that helps understand this better.

**Review by datapythonista — COMMENTED** [review]
Really nice, I added couple of suggestions, but great addition.

File: doc/source/user_guide/pyarrow.rst (hunk 2 above)
Comment (mroeschke) [reply doc/source/user_guide/pyarrow.rst:14]: Sure that makes sense. Mentioned that it helps interoperability with other libraries that implement the Arrow specification

File: doc/source/user_guide/pyarrow.rst (hunk 3 above)
Comment (mroeschke) [reply doc/source/user_guide/pyarrow.rst:92]: Good idea. Added a sentence that these are some examples that use PyArrow compute functions

**Review by datapythonista — APPROVED** [review]
Really nice, looks perfect!

**Review by phofl — APPROVED** [review]

File: doc/source/user_guide/pyarrow.rst [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+        df_pyarrow = pd.read_csv(data, use_nullable_dtypes=True)
+    df_pyarrow.dtypes
+
+To simplify specifying ``use_nullable_dtypes=True`` in several functions, you can set a global option ``nullable_dtypes``
+to ``True``. You will still need to set the global configuration option ``"mode.dtype_backend"`` to ``pyarrow``.
+
+.. ipython:: python
+
+    pd.set_option("mode.dtype_backend", "pyarrow")
+    pd.options.mode.nullable_dtypes = True
```
Comment (phofl) [inline doc/source/user_guide/pyarrow.rst:156]: I think this has side effect, I guess we need to reset to False after this chapter is over (ci failure seems to be caused by this)

Comment (mroeschke) [reply doc/source/user_guide/pyarrow.rst]: Will convert to a code block instead

Comment (phofl) [reply doc/source/user_guide/pyarrow.rst]: That works as well, thx

**Pull request merged**